src/control/cmd/daos_server/storage.go

```diff
@@ -30,6 +32,8 @@ type storagePrepareCmd struct {
 	scs *server.StorageControlService
 	logCmd
 	commands.StoragePrepareCmd
+	HelperLogFile string `short:"l" long:"helper-log-file" description:"Log debug from daos_admin binary."`
+	EnableVMD     bool   `long:"enable-vmd" description:"Additionally try to prepare any discovered VMD NVMe devices."`
```

Comment:
Yes, I think VMD should be enabled by default, what do you think? @sydneyvanda

Reply:
@tanabarr @mjmac  @NiuYawei I think enabling VMD by default could be very tricky. If we want to go that direction, I think we have to only unbind the VMD addresses that are provided in the bdev_include list (if they are present on the system) and not unbind all the VMD addresses by default. I think there could be a large majority of users that don't really understand VMD, and if it is enabled by default, then all of there drives could be unbound by default, some with valid VMD addresses, and some behind multiple VMD addresses, and it might cause some confusion. I liked the idea of having to manually turn on VMD support, and with that provide a VMD address that you intend to use (knowing there are backing NVMe SSDs). 

However if we want to enable by default, I think there are a few more sanity checks that need to happen before any unbinding:
1- Continue to check if VMD addresses are present on the system, otherwise disable VMD
2- Use $ll /sys/block/ output to see initial VMD mappings of the NVMe SSDs. Skip any VMD addresses that are present that have no backing NVMe SSDs. We could also use the SPDK lsvmd utility for this but would be more work. 
3- Unbind VMD first, then NVMe. This will prevent any 100*: prefix devices from being unbound as an NVMe SSD before VMD address is unbound.
(all the above is of course assuming there is no bdev_include list provided, and should be simplified if there is one)

Reply:
@sydneyvanda: Thanks for your input on this. I understand that enabling VMD by default now might potentially cause some problems as we learn more about how it works and how it will be used. From a long-term perspective, though, do you think it will be the norm?

My primary concern right now is that we're consistent throughout the codebase. If we think that VMD will always be something that should optionally be enabled, then we should change the code to reflect that. If we think that it will be eventually be the norm and optionally disabled, then we should make the code reflect that position. It's simple enough to make the default CLI option enabled or disabled -- I'm just trying to avoid having ripples throughout the codebase due to changes in the default value.

Reply:
makes sense, honestly I'm not sure how popular VMD use will be in the future (for normal DAOS customers). The main reason why we are using VMD is for the LED control component as that will solve the problem of quickly identifying any failed SSDs in a large cluster of thousands of SSDs so those drives can be hotplugged (which is something ANL requires specifically). Doing any other sort of mapping would be difficult and not straightforward, so this was a good solution. 
What I am finding is many VMD users are really VROC users (which is the RAID component linked with VMD), however DAOS does not use that as we have our own solution for that. We are just using the LED capabilities in DAOS in part with the NVMe hotplug feature.

Reply:
OK, so it sounds like VMD support should not be enabled by default, and the codebase should be structured to reflect that VMD is an optional alternative way of doing things. @tanabarr, please update your PR to make this consistent throughout the codebase. I'd really prefer to do it now so that the next test build (which is currently the last scheduled test build) has the code as close to the way we're planning on releasing it.

Reply:
done